CMakeLists.txt

```diff
@@ -32,17 +32,25 @@ endif()
 # ------------------------------------------------------------------------------
 find_package(Threads REQUIRED)
 find_package(OpenMP REQUIRED)
-find_package(BLAS REQUIRED)
-if (BLAS_FOUND)
-    message(STATUS "BLAS found: ${BLAS_LIBRARIES}")
-else ()
-    message(FATAL_ERROR "BLAS not found")
-endif ()
-
 
 set(FETCHCONTENT_QUIET OFF)
 include(FetchContent)
 
+# Fetch and build OpenBLAS
+FetchContent_Declare(
+  OpenBLAS
+  GIT_REPOSITORY https://github.com/xianyi/OpenBLAS.git
+  GIT_TAG v0.3.29
+)
+
+# Set OpenBLAS build options
+set(NOFORTRAN ON CACHE BOOL "Disable Fortran" FORCE)
+set(BUILD_WITHOUT_LAPACK OFF CACHE BOOL "Build without LAPACK" FORCE)
+set(USE_THREAD ON CACHE BOOL "Use threading" FORCE)
```

Comment:
In general, BLAS to map to OpenBLAS on all platforms is a probably a good idea. Agreed 🤝